linkerd/app/gateway/src/discover.rs

```diff
@@ -71,7 +64,14 @@ impl Gateway {
                     // names that are not Services, so we will get a `NotFound`
                     // error if we looked up a pod DNS name. In this case, we
                     // will synthesize a default policy.
-                    Err(error) if is_not_found(&error) => tracing::debug!("Policy not found"),
+                    Err(error) if errors::has_grpc_status(&error, tonic::Code::NotFound) =>
+                        tracing::debug!("Policy not found"),
+                    // Earlier versions of the Linkerd control plane (e.g.
+                    // 2.12.x) will return `Unimplemented` for requests to the
+                    // OutboundPolicy API. Log a warning and synthesize a policy
+                    // for backwards compatibility.
+                    Err(error) if errors::has_grpc_status(&error, tonic::Code::Unimplemented) =>
+                        tracing::warn!("Policy controller returned `Unimplemented`, the control plane may be out of date."),
```

Comment:
This could generate lots o
```suggestion
                    // OutboundPolicy API.
                    Err(error) if errors::has_grpc_status(&error, tonic::Code::Unimplemented) =>
                        tracing::debug!("Policy controller returned `Unimplemented`, the control plane may be out of date."),
```